src/amber/router/route.cr

```diff
@@ -49,7 +49,11 @@ module Amber
       {result, params}
     end
 
-    def match?(controller, action)
+    def match?(controller : Class, action : Symbol)
```

Comment:
This is separated into 2 method so we don't have to perform checks or substitutions for `/Controller$/`